metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/dereference/DereferencerImpl.java

```diff
@@ -0,0 +1,123 @@
+package eu.europeana.enrichment.rest.client.dereference;
+
+import static eu.europeana.metis.utils.ExternalRequestUtil.retryableExternalRequestForNetworkExceptions;
+
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.EnrichmentBaseWrapper;
+import eu.europeana.enrichment.api.external.model.EnrichmentResultList;
+import eu.europeana.enrichment.rest.client.enrichment.EnrichmentClient;
+import eu.europeana.enrichment.rest.client.exceptions.DereferenceException;
+import eu.europeana.enrichment.utils.DereferenceUtils;
+import eu.europeana.enrichment.utils.EntityMergeEngine;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.client.HttpClientErrorException.BadRequest;
+
+public class DereferencerImpl implements Dereferencer{
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Dereferencer.class);
+
+  private final EntityMergeEngine entityMergeEngine;
+  private final EnrichmentClient enrichmentClient;
+  private final DereferenceClient dereferenceClient;
+
+  public DereferencerImpl(EntityMergeEngine entityMergeEngine,
```

Comment:
This contradicts the Builder pattern where the user can actually create this object by using this constructor(normally the constructor should be private)..
Are we allowing for that?
If yes then perhaps the Builder naming should change to something else to avoid confusion..

(This looks more like a factory pattern to me)

Reply:
I changed the name of the class just like we discussed